react/components/Collapsible.js

```diff
@@ -1,51 +1,67 @@
-import React from 'react'
-import PropTypes from 'prop-types'
+import React, { Fragment } from 'react'
 import classNames from 'classnames'
-import withScrollAnimation from './ScrollAnimation'
-import { Border } from './Helpers'
+import { useSpring, animated, config as springPresets } from 'react-spring'
 import useDevice from '../hooks/useDevice'
+import useScrollDirection from '../hooks/useScrollDirection'
+import Border from './Helpers/Border'
+import { collapsible, lean } from '../defaults'
 
-import header from '../store-header.css'
+import styles from '../store-header.css'
 
-const Collapsible = ({
-  children,
-  leanMode,
-  animation,
-  didAnimate,
-  onAnimate,
-}) => {
+/**
+ * Represents a collapsible part of the header
+ */
+const Collapsible = ({ children, leanMode, collapsibleAnimation }) => {
   const { desktop } = useDevice()
-  onAnimate(didAnimate)
+  const { scroll, scrollingUp } = useScrollDirection()
+  const {
+    onScroll,
+    anchor,
+    always,
+    from,
+    to,
+    preset,
+    config,
+  } = collapsibleAnimation
+
+  let animationTrigger = true
+
+  if (onScroll) {
+    const animateWhen = scroll < anchor
+    const elastic = always ? scrollingUp : false
+    animationTrigger = animateWhen || elastic
+  }
+
+  const props = useSpring({
+    config: preset ? springPresets[preset] : config,
+    height: animationTrigger ? from : to,
```

Comment:
animating `height` triggers a re-layout every frame, which is a bit expensive on slower CPUs (such as mobile devices). prefer animating the `transform` instead (e.g. `translate3d`, `scale`) which just triggers a `composite` event, which is done on the GPU

[more info here](https://developers.google.com/web/fundamentals/performance/rendering/avoid-large-complex-layouts-and-layout-thrashing#avoid_forced_synchronous_layouts), [TLDR cheatsheet here](https://devhints.io/layout-thrashing)

Reply:
Nice catch 🎯